<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Task 3 - Dynamic Network Visualization &mdash; Cork Oak Use Case  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Task 4 - Ensembl Plants Query" href="task4.html" />
    <link rel="prev" title="Task 2 - Metabolic Pathway Visualization" href="task2.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Cork Oak Use Case
              <img src="../_static/logo.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Tasks</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="task1.html">Task 1 - Plant Data Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="task2.html">Task 2 - Metabolic Pathway Visualization</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Task 3 - Dynamic Network Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="task4.html">Task 4 - Ensembl Plants Query</a></li>
</ul>
<p class="caption"><span class="caption-text">Tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tool1_mercator.html">Mercator4 Webtool</a></li>
<li class="toctree-l1"><a class="reference internal" href="tool2_mapman.html">MapMan Desktop App</a></li>
<li class="toctree-l1"><a class="reference internal" href="tool3_dinar.html">DiNAR App</a></li>
</ul>
<p class="caption"><span class="caption-text">References</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="references.html">References</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Cork Oak Use Case</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Task 3 - Dynamic Network Visualization</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/source/task3.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="task-3-dynamic-network-visualization">
<span id="task3-label"></span><h1>Task 3 - Dynamic Network Visualization<a class="headerlink" href="#task-3-dynamic-network-visualization" title="Permalink to this headline"></a></h1>
<div class="line-block">
<div class="line"><a class="reference external" href="https://nib-si.shinyapps.io/DiNAR/">DiNAR</a> is a tool used to examine omics datasets in a prior knowledge context. This is achieved through the dynamic representation of biological data  over multiple timepoints or conditions in a network format.</div>
<div class="line">The current task will guide the user on how to import and analyse networks into DiNAR and obtain outputs like <a class="reference internal" href="#dinar-example-label"><span class="std std-ref">these</span></a>.</div>
<div class="line">To start using DiNAR, two options are provided in this usecase to prepare a gene/protein network with specific requirements:</div>
<div class="line"><br /></div>
<div class="line">* <a class="reference internal" href="#option1-label"><span class="std std-ref">Option 1</span></a> - Using the provided <code class="docutils literal notranslate"><span class="pre">Cytoscape_to_Dinar.R</span></code> script to prepare the network.</div>
<div class="line">* <a class="reference internal" href="#option2-label"><span class="std std-ref">Option 2</span></a> - Using the DiNAR preprocessing subApp to prepare the network.</div>
</div>
<section id="option-1-importing-a-custom-network-into-dinar-with-cytoscape-to-dinar">
<span id="option1-label"></span><h2>Option 1 - Importing a Custom Network into DiNAR with <code class="docutils literal notranslate"><span class="pre">Cytoscape_to_Dinar</span></code><a class="headerlink" href="#option-1-importing-a-custom-network-into-dinar-with-cytoscape-to-dinar" title="Permalink to this headline"></a></h2>
<div class="line-block">
<div class="line">This option allows the user to provide simple node and edge tables to import into DiNAR. For example, these can be obtained from Cytoscape (access this <a class="reference internal" href="#cytoscape-label"><span class="std std-ref">section</span></a> to learn how).</div>
<div class="line">Additional settings are provided to enrich the network with the <code class="docutils literal notranslate"><span class="pre">corkoak.results</span></code> obtained in <a class="reference internal" href="task1.html#task1-label"><span class="std std-ref">these</span></a>, and usefull information from the Ensembl Plants database.</div>
</div>
<div class="line-block">
<div class="line">For this option, the initial node and edge tables should have the following format(s):</div>
</div>
<ul class="simple">
<li><p>Node Table Example (.csv):</p></li>
</ul>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_45155</p></td>
</tr>
<tr class="row-even"><td><p>CFP56_55251</p></td>
</tr>
<tr class="row-odd"><td><p>…</p></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This column is mandatory in the first position and must have gene/protein IDs. Additional columns can be present.</p>
</div>
<ul class="simple">
<li><p>Edge Table Example 1 (.csv):</p></li>
</ul>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 29%" />
<col style="width: 29%" />
<col style="width: 43%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID1</p></th>
<th class="head"><p>GeneID2</p></th>
<th class="head"><p>reactionType</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
<td><p>CFP56_45155</p></td>
<td><p>coexpressed_with</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_37705</p></td>
<td><p>CFP56_55251</p></td>
<td><p>coexpressed_with</p></td>
</tr>
<tr class="row-even"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<ul class="simple">
<li><p>Edge Table Example 2 (.csv):</p></li>
</ul>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 57%" />
<col style="width: 43%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>name</p></th>
<th class="head"><p>interaction</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705 (interacts with) CFP56_45155</p></td>
<td><p>interacts with</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_37705 (interacts with) CFP56_55251</p></td>
<td><p>interacts with</p></td>
</tr>
<tr class="row-even"><td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <code class="docutils literal notranslate"><span class="pre">corkoak_node.csv</span></code> and <code class="docutils literal notranslate"><span class="pre">corkoak_edge.csv</span></code>, present in the <strong>CorkOak_UseCase_Data-main</strong> folder, already conform to this format.</p>
</div>
<ol class="arabic" id="startscript-label">
<li><div class="line-block">
<div class="line">Open the command line (Command Prompt) and run <code class="docutils literal notranslate"><span class="pre">Cytoscape_to_Dinar.R</span></code></div>
<div class="line">For a detailed usage of this script, go to <a class="reference internal" href="#script-usage-label"><span class="std std-ref">Cytoscape_to_Dinar Usage</span></a>.</div>
</div>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="c1">#Don&#39;t forget to replace the PathToFile section with the Directory where the CorkOak_UseCase_Data-main Folder is located</span>
<span class="n">Rscript</span> <span class="n">PathToFile</span><span class="o">/</span><span class="n">Cytoscape_to_Dinar.R</span> <span class="o">-</span><span class="n">n</span> <span class="n">corkoak_node.csv</span> <span class="o">-</span><span class="n">e</span> <span class="n">corkoak_edge.csv</span> <span class="o">-</span><span class="n">c</span> <span class="n">corkoak.results</span> <span class="o">-</span><span class="n">o</span> <span class="n">corkoak_dinar</span>
</pre></div>
</div>
</li>
</ol>
<p>After running the script, the output files should look like the following:</p>
<ul class="simple">
<li><p>DiNAR Ready Node Table <code class="docutils literal notranslate"><span class="pre">corkoak_dinar_node.txt</span></code>:</p></li>
</ul>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 7%" />
<col style="width: 11%" />
<col style="width: 15%" />
<col style="width: 15%" />
<col style="width: 7%" />
<col style="width: 15%" />
<col style="width: 15%" />
<col style="width: 7%" />
<col style="width: 7%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID</p></th>
<th class="head"><p>shortName</p></th>
<th class="head"><p>shortDescription</p></th>
<th class="head"><p>MapManBIN</p></th>
<th class="head"><p>clusterID</p></th>
<th class="head"><p>x</p></th>
<th class="head"><p>y</p></th>
<th class="head"><p>clusterSimplifiedNodeDegree</p></th>
<th class="head"><p>expressed</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
<td><p>TPX2</p></td>
<td><p>protein TPX2</p></td>
<td><p>35.2_not classified</p></td>
<td><p>1</p></td>
<td><p>13.2944396706271</p></td>
<td><p>13.9080090717365</p></td>
<td><p>2</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_45155</p></td>
<td><p>NAC43</p></td>
<td><p>NAC domain-containing protein 43</p></td>
<td><p>28.2.1.11.2_transcriptional regulator *(NST1/NST2)</p></td>
<td><p>1</p></td>
<td><p>6.39534301750403</p></td>
<td><p>20.3626647889864</p></td>
<td><p>20</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p>CFP56_55251</p></td>
<td><p>COBRA4</p></td>
<td><p>COBRA-like protein 4</p></td>
<td><p>21.1.2.2_regulatory protein *(COB) of cellulose-hemicellulose network assembly</p></td>
<td><p>1</p></td>
<td><p>22.5121051761507</p></td>
<td><p>11.7050783234164</p></td>
<td><p>12</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<ul class="simple">
<li><p>DiNAR Ready Edge Table <code class="docutils literal notranslate"><span class="pre">corkoak_dinar_edge.txt</span></code>:</p></li>
</ul>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 18%" />
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 12%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID1</p></th>
<th class="head"><p>GeneID2</p></th>
<th class="head"><p>reactionType</p></th>
<th class="head"><p>clusterID_geneID1</p></th>
<th class="head"><p>clusterID_geneID2</p></th>
<th class="head"><p>clusterSimplifiedNodeDegree_geneID1</p></th>
<th class="head"><p>clusterSimplifiedNodeDegree_geneID2</p></th>
<th class="head"><p>exists</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
<td><p>CFP56_45155</p></td>
<td><p>coexpressed_with</p></td>
<td><p>1</p></td>
<td><p>1</p></td>
<td><p>2</p></td>
<td><p>20</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_37705</p></td>
<td><p>CFP56_55251</p></td>
<td><p>coexpressed_with</p></td>
<td><p>1</p></td>
<td><p>1</p></td>
<td><p>2</p></td>
<td><p>12</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<p>With the tables ready, jump to the following <a class="reference internal" href="#main-dinar-label"><span class="std std-ref">Main DiNAR section</span></a>.</p>
</section>
<section id="option-2-preparing-a-custom-network-into-dinar-from-other-sources">
<span id="option2-label"></span><h2>Option 2 - Preparing a Custom Network into DiNaR (from other sources)<a class="headerlink" href="#option-2-preparing-a-custom-network-into-dinar-from-other-sources" title="Permalink to this headline"></a></h2>
<p>Alternatively, to import a custom network into <a class="reference external" href="https://nib-si.shinyapps.io/DiNAR/">DiNAR</a> and avoid the <code class="docutils literal notranslate"><span class="pre">Cytoscape_to_Dinar.R</span></code> script, a node and edge tables should be supplied following a specified tabular structure, which must be firstly processed with the <a class="reference external" href="https://nib-si.shinyapps.io/pre-processing/">DiNaR preprocessing subApp</a>. The table formats necessary for this option are the following:</p>
<ul class="simple">
<li><p>Node Table Example (before DiNAR preprocessing):</p></li>
</ul>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID</p></th>
<th class="head"><p>shortName</p></th>
<th class="head"><p>shortDescription</p></th>
<th class="head"><p>MapManBin</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_45155</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
</tr>
<tr class="row-even"><td><p>CFP56_55251</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
</tr>
<tr class="row-odd"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line">The first column is the only one required to be filled (either with gene or protein IDs). The other columns are required but may be left empty (using “-“).</div>
<div class="line">Additional columns can also be present.</div>
</div>
<ul class="simple">
<li><p>Edge Table Example (before DiNAR preprocessing):</p></li>
</ul>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 29%" />
<col style="width: 29%" />
<col style="width: 43%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID1</p></th>
<th class="head"><p>GeneID2</p></th>
<th class="head"><p>reactionType</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
<td><p>CFP56_45155</p></td>
<td><p>coexpressed_with</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_37705</p></td>
<td><p>CFP56_55251</p></td>
<td><p>coexpressed_with</p></td>
</tr>
<tr class="row-even"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<ol class="arabic simple">
<li><p>Open the <a class="reference external" href="https://nib-si.shinyapps.io/pre-processing/">DiNaR preprocessing subApp</a> and select the <strong>tables</strong> option tab</p></li>
<li><p>Click <strong>Choose Nodes File</strong> and select a input node table</p></li>
<li><p>Click <strong>Choose Edges File</strong> and select a input edge table</p></li>
<li><p>In the Nodes Tab, click download (button under the plot visualization). It should look like the following:</p></li>
</ol>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 9%" />
<col style="width: 13%" />
<col style="width: 17%" />
<col style="width: 17%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID</p></th>
<th class="head"><p>shortName</p></th>
<th class="head"><p>shortDescription</p></th>
<th class="head"><p>MapManBIN</p></th>
<th class="head"><p>clusterID</p></th>
<th class="head"><p>x</p></th>
<th class="head"><p>y</p></th>
<th class="head"><p>clusterSimplifiedNodeDegree</p></th>
<th class="head"><p>expressed</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>1</p></td>
<td><p>13.2944396706271</p></td>
<td><p>13.9080090717365</p></td>
<td><p>2</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_45155</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>1</p></td>
<td><p>6.39534301750403</p></td>
<td><p>20.3626647889864</p></td>
<td><p>20</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p>CFP56_55251</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>1</p></td>
<td><p>22.5121051761507</p></td>
<td><p>11.7050783234164</p></td>
<td><p>12</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<ol class="arabic simple" start="5">
<li><p>In the Edges Tab, click download (button under the plot visualization). It should look like the following:</p></li>
</ol>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 18%" />
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 12%" />
<col style="width: 12%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>GeneID1</p></th>
<th class="head"><p>GeneID2</p></th>
<th class="head"><p>reactionType</p></th>
<th class="head"><p>clusterID_geneID1</p></th>
<th class="head"><p>clusterID_geneID2</p></th>
<th class="head"><p>clusterSimplifiedNodeDegree_geneID1</p></th>
<th class="head"><p>clusterSimplifiedNodeDegree_geneID2</p></th>
<th class="head"><p>exists</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CFP56_37705</p></td>
<td><p>CFP56_45155</p></td>
<td><p>coexpressed_with</p></td>
<td><p>1</p></td>
<td><p>1</p></td>
<td><p>2</p></td>
<td><p>20</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>CFP56_37705</p></td>
<td><p>CFP56_55251</p></td>
<td><p>coexpressed_with</p></td>
<td><p>1</p></td>
<td><p>1</p></td>
<td><p>2</p></td>
<td><p>12</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<section id="video-guide-for-dinar-preprocessing-subapp">
<h3><strong>Video guide for</strong> <a class="reference external" href="https://nib-si.shinyapps.io/pre-processing/">DiNaR preprocessing subApp</a><a class="headerlink" href="#video-guide-for-dinar-preprocessing-subapp" title="Permalink to this headline"></a></h3>
<iframe width="560" height="315" src="https://www.youtube.com/embed/1mbPwHR12RU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe><div class="line-block">
<div class="line"><br /></div>
</div>
<p>With the tables ready with this alternative method, we can then import them into DiNAR.</p>
<div class="line-block" id="main-dinar-label">
<div class="line"><br /></div>
</div>
</section>
</section>
<section id="importing-a-custom-network-into-the-main-dinar-app">
<h2>Importing a Custom Network into the Main DiNaR App<a class="headerlink" href="#importing-a-custom-network-into-the-main-dinar-app" title="Permalink to this headline"></a></h2>
<ol class="arabic simple">
<li><p>Go to the <a class="reference external" href="https://nib-si.shinyapps.io/DiNAR/">DiNaR App</a></p></li>
<li><p>In <strong>select network</strong>, choose <strong>Custom network</strong></p></li>
<li><p>In <strong>Upload nodes table</strong>, select the <code class="docutils literal notranslate"><span class="pre">corkoak_dinar_node.txt</span></code> file</p></li>
<li><p>In <strong>Upload edges table</strong>, select the <code class="docutils literal notranslate"><span class="pre">corkoak_dinar_edge.txt</span></code> file</p></li>
<li><p>In <strong>Upload Experimental data files</strong>, select the <code class="docutils literal notranslate"><span class="pre">corkoak_LogFC_April.txt</span></code>, <code class="docutils literal notranslate"><span class="pre">corkoak_LogFC_June.txt</span></code> and <code class="docutils literal notranslate"><span class="pre">corkoak_LogFC_July.txt</span></code> files</p></li>
<li><p>Confirm columns containing gene IDs (geneID), statistical significance (adj.P.Val) and the logFC values (LogFC_”month”)</p></li>
<li><p>Click <strong>Proceed!</strong></p></li>
</ol>
<section id="exploring-dinar-network-visualization-options">
<h3>Exploring DiNaR network visualization options<a class="headerlink" href="#exploring-dinar-network-visualization-options" title="Permalink to this headline"></a></h3>
<ol class="arabic simple">
<li><p>Click the <strong>Cluster Information</strong> Tab, and select the <strong>MapMan BINs</strong> option</p></li>
</ol>
<figure class="align-default">
<a class="reference internal image-reference" href="../_images/Export_Node_Table.png"><img alt="../_images/Export_Node_Table.png" src="../_images/Export_Node_Table.png" style="width: 594.8000000000001px; height: 401.0px;" /></a>
</figure>
<div class="line-block">
<div class="line">This image highlights the most populated Bins throughout the Input Network, with the X - cell wall being the most prevalent for our example.</div>
<div class="line">Search and Filtering options under the bin and description columns can help identify processes of interest occurrying in biological networks.</div>
</div>
</section>
<section id="creating-a-dynamic-network-visualization">
<h3>Creating a dynamic network visualization<a class="headerlink" href="#creating-a-dynamic-network-visualization" title="Permalink to this headline"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Larger amounts of experimental data allows the user to create longer and richer dynamic network visualizations in DiNAR.</p>
</div>
<ol class="arabic simple">
<li><p>Click the Differential expression per cluster Tab, and select the Dynamic-animatoR option</p></li>
<li><p>Scroll down and, under the <strong>Select speed</strong> option, click on the video speed <strong>1</strong></p></li>
</ol>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Different speed settings can be used to obtained visualizations which go faster (speed <strong>1</strong>) or slower (speed <strong>0.1</strong>) throughout multiple conditions/timepoints.</p>
</div>
<ol class="arabic simple" start="3">
<li><p>Click <strong>Create dynamic.html</strong></p></li>
<li><p>After a couple seconds (depending on network size), click <strong>Download dynamic.html</strong></p></li>
<li><p>Open the .html file and click the <img alt="start" src="../_images/start_symbol.PNG" /> starting video symbol</p></li>
</ol>
<ul class="simple">
<li><p>Cork Oak Dynamic Network Visualization:</p></li>
</ul>
<iframe width="560" height="315" src="../_static/SeasonalGrowth.html" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe><div class="line-block">
<div class="line"><br /></div>
</div>
<div class="line-block">
<div class="line">It is clear to notice, through the visualization of the generated video, that all cork oak genes present in the example network have a concerted increase in gene expression towards the month of July (summer). Together with the information inferred in the previous task, these observations highlight a steady gene upregulation in cork oak as this species seasonal growth includes gene upregulation towards lignin biosynthesis and tree growth processes during summer.</div>
</div>
<div class="line-block">
<div class="line">Larger and more complex networks, with mixing gene downregulation and upregulation, and additional curated data, can be visualized in DiNAR to observe time-specific phenomena in a clear and intuitive manner.</div>
</div>
</section>
<section id="video-guide-for-dinar-app">
<h3><strong>Video guide for</strong> <a class="reference external" href="https://nib-si.shinyapps.io/DiNAR/">DiNaR App</a><a class="headerlink" href="#video-guide-for-dinar-app" title="Permalink to this headline"></a></h3>
<iframe width="560" height="315" src="https://www.youtube.com/embed/1mbPwHR12RU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe><div class="line-block" id="cytoscape-label">
<div class="line"><br /></div>
</div>
<p>Now, if ready, the user can move into the next and final (<a class="reference internal" href="task4.html#task4-label"><span class="std std-ref">Task 4 - Ensembl Plants Query</span></a>), which will help in the retrieval of information from the Ensembl Plants database from the command line.</p>
</section>
<section id="getting-node-and-edge-tables-from-cytoscape">
<h3>Getting Node and Edge Tables from Cytoscape<a class="headerlink" href="#getting-node-and-edge-tables-from-cytoscape" title="Permalink to this headline"></a></h3>
<p><a class="reference external" href="https://cytoscape.org/">Cytoscape</a> is a widely used tool for network analysis and visualization. This section will guide the user on how to obtain node/edge tables to be imported into DiNAR using the <code class="docutils literal notranslate"><span class="pre">Cytoscape_to_Dinar.R</span></code> script.</p>
<ol class="arabic simple">
<li><p>In Cytoscape, select the desired network</p></li>
<li><p>Click the <strong>Node Table</strong> tab</p></li>
<li><p>Select the <strong>Export Table to File…</strong> option, saving it in the default (.csv) format</p></li>
</ol>
<figure class="align-default">
<a class="reference internal image-reference" href="../_images/Export_Node_Table.png"><img alt="../_images/Export_Node_Table.png" src="../_images/Export_Node_Table.png" style="width: 594.8000000000001px; height: 401.0px;" /></a>
</figure>
<ol class="arabic simple" start="4">
<li><p>Click the <strong>Edge Table</strong> tab</p></li>
<li><p>Select the <strong>Export Table to File…</strong> option, saving it in the default (.csv) format</p></li>
</ol>
<figure class="align-default">
<a class="reference internal image-reference" href="../_images/Export_Edge_Table.png"><img alt="../_images/Export_Edge_Table.png" src="../_images/Export_Edge_Table.png" style="width: 596.0px; height: 399.6px;" /></a>
</figure>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Both node and edge tables can be retrieved from several databases or retrieved from other network visualization tools, as long the structure remains DiNAR compatible.</p>
</div>
<div class="line-block" id="script-usage-label">
<div class="line">With these tables from Cytoscape, the user can resume the workflow to prepare and import them into DiNAR <a class="reference internal" href="#startscript-label"><span class="std std-ref">here</span></a>.</div>
<div class="line"><br /></div>
</div>
</section>
<section id="rscript-usage">
<h3>Rscript Usage:<a class="headerlink" href="#rscript-usage" title="Permalink to this headline"></a></h3>
<blockquote>
<div><div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="n">Rscript</span> <span class="n">PathToFile</span><span class="o">/</span><span class="n">Cytoscape_to_Dinar.R</span> <span class="p">[</span><span class="o">-</span><span class="n">n</span><span class="p">]</span> <span class="n">node_table.csv</span> <span class="p">[</span><span class="o">-</span><span class="n">e</span><span class="p">]</span> <span class="n">edge_table</span> <span class="o">-</span><span class="n">c</span> <span class="n">corkoak.results</span> <span class="o">-</span><span class="n">e</span> <span class="n">species_name</span> <span class="o">-</span><span class="n">o</span> <span class="n">output_name</span>
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">[-n]</span> <span class="pre">node</span> <span class="pre">table</span></code> - Input a Node Table (.csv, comma-separated)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">[-e]</span> <span class="pre">edge</span> <span class="pre">table</span></code> - Input a Edge Table (.csv, comma-separated)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">[-c]</span> <span class="pre">custom</span> <span class="pre">annotation</span></code> - Input gene/protein annotations obtained in Mercator4 / mercator.results (optional)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">[-e]</span> <span class="pre">ensemblplants</span></code> - Retrieve short names, short descriptions and orthologs from the Ensembl Plants database (optional)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">[-o]</span> <span class="pre">output</span></code> - Specify a name for the output node and edge tables</p></li>
</ul>
</div></blockquote>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Do not forget to specify the path for the directory where this script is located (CorkOak_UseCase_Data-main).
This should also be the folder with the corkoak_node.csv, corkoak_edge.csv and corkoak.results.txt files.</p>
</div>
<div class="line-block" id="dinar-example-label">
<div class="line"><br /></div>
</div>
</section>
<section id="dinar-output-examples">
<h3>DiNAR Output Examples<a class="headerlink" href="#dinar-output-examples" title="Permalink to this headline"></a></h3>
<p>The following visualizations were obtained from a <em>Arabidopsis thaliana</em> network showcasing gene expression in multiple timepoints after <em>Pseudomonas syringae</em> infection (described <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6117943/">here</a>).</p>
<ul class="simple">
<li><p>Example 1 - Interactive Visualization Option (.html format)</p></li>
</ul>
<p>This option allows the user to click nodes and edges of the network to check their summarized information, and highlights the expression tendencies of specific genes throughtout the course of infection response.</p>
<iframe width="700" height="400" src="../_static/LKN_QuickAppTest(Ath123)_clu_1_deg_0_tp_1NDTV.html" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe><div class="line-block">
<div class="line"><br /></div>
</div>
<ul class="simple">
<li><p>Example 2 - Video Option (.mp4 format)</p></li>
</ul>
<iframe width="560" height="315" src="https://www.youtube.com/embed/f-mhm-MqgSU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe><div class="line-block">
<div class="line"><br /></div>
</div>
<div class="line-block">
<div class="line">This concludes Task 3 of the current usecase.</div>
<div class="line">The next and final <a class="reference internal" href="task4.html#task4-label"><span class="std std-ref">Task 4 - Ensembl Plants Query</span></a> will guide the user on how to retrieve information from the Ensembl Plants database, independently of the <code class="docutils literal notranslate"><span class="pre">Cytoscape_to_Dinar.R</span></code> script.</div>
<div class="line"><br /></div>
</div>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="task2.html" class="btn btn-neutral float-left" title="Task 2 - Metabolic Pathway Visualization" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="task4.html" class="btn btn-neutral float-right" title="Task 4 - Ensembl Plants Query" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Hugo Rodrigues.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>